app/frontend/wallet/shelley-wallet.ts

```diff
@@ -200,7 +193,9 @@ const ShelleyBlockchainExplorer = (config) => {
       throw NamedError('NetworkError', e.message)
     }
     const poolArray = JSON.parse(await response.text())
+    // eslint-disable-next-line no-sequences
     const validStakepools = poolArray.reduce((dict, el) => ((dict[el.pool_id] = {...el}), dict), {})
+    // eslint-disable-next-line no-sequences
     const ticker2Id = poolArray.reduce((dict, el) => ((dict[el.ticker] = el.pool_id), dict), {})
     return {validStakepools, ticker2Id}
```

Comment:
personally, I would prefer to return a single array of stake pools and store it like that in the state as well, avoiding having duplicate related variables in it which may accidentally fall out of sync. And then implement helpers to retrieve that information from that array.

Btw, accessing/setting object properties by `[propName]` may lead to weird glitches, if some stake pool happens to be named like a JS object prototype property, imagine this code:

```
let x = {"a": 3}
console.log(x["constructor") // outputs a function()
```

Reply:
The point of two dictionaries was to be able to choose a stake pool by ID and TICKER, and do both of these in O(1). But I agree we should create a separate issue for this. For example, it seems a ticker is not unique identifier for a stake pool, so this whole searching by ticker might not be a good idea.